src/lib/Scenes/Home/Components/ForYou/Components/ArtworkCarouselHeader.tsx

```diff
@@ -108,8 +108,6 @@ class ArtworkCarouselHeader extends Component<Props & RelayPropsWorkaround, Stat
     ARTemporaryAPIModule.setFollowArtistStatus(!this.state.following, context.artist.id, (error, following) => {
       if (error) {
         console.error("ArtworkCarouselHeader.tsx", error)
-      } else {
```

Comment:
Just so I understand what the bug was - this code was executing even if the artist didn't exist, and the above check for `HomePageModuleContextRelatedArtist` ensures the rail definitely has artist details?

Reply:
This method would execute, yeah, but the crash would occur on [line 102](https://github.com/artsy/emission/pull/966/files#diff-b3db040e2aa0a16c5d8a4f98d02c0ae5R102) where `context.artist` was `undefined` when this module represented a ‘followed artist’, as the data for that was not being queried for. That query is now added to the fragment.